gapic/src/main/com/google/gapid/widgets/TabComposite.java

```diff
@@ -0,0 +1,1173 @@
+        }
+      }
+    }
+
+    @Override
+    protected Hover find(Group parent, int index, int mx, int my, boolean includeTrailing) {
+      for (int i = 0; i < children.size(); i++) {
+        Element child = children.get(i);
+        if (mx >= child.x && mx < child.x + child.w &&
+            my >= child.y && my < child.y + child.h) {
+          return child.find(this, i, mx, my, includeTrailing);
+        }
+      }
+      return Hover.NONE;
+    }
+
+    protected abstract void moveSeparator(int index, int sx, int sy);
+
+    @Override
+    protected MergeState merge() {
+      for (ListIterator<Element> it = children.listIterator(); it.hasNext(); ) {
+        Element current = it.next();
+        MergeState state = current.merge();
+        if (state == MergeState.REMOVE) {
+          it.remove();
+        } else if (state == MergeState.DO_NOTHING) {
+          // Do nothing.
+        } else {
+          state.replacement.weight = current.weight;
+          it.set(state.replacement);
+        }
+      }
+
+      switch (children.size()) {
+        case 0: return MergeState.REMOVE;
+        case 1: return MergeState.replace(children.get(0));
+        default: return MergeState.DO_NOTHING;
+      }
+    }
+  }
+
+  private static enum Location {
+    Left, Right, Top, Bottom;
+
+    public Rectangle highlight(Folder f) {
+      switch (this) {
+        case Left:   return new Rectangle(f.x, f.y, f.w / 3, f.h);
+        case Right:  return new Rectangle(f.x + 2 * f.w / 3, f.y, f.w / 3, f.h);
+        case Top:    return new Rectangle(f.x, f.y, f.w, f.h / 3);
+        case Bottom: return new Rectangle(f.x, f.y + 2 * f.h / 3, f.w, f.h / 3);
+        default: throw new AssertionError();
+      }
+    }
+  }
+
+  private class HorizontalGroup extends Group {
+    public HorizontalGroup(int weight) {
+      super(weight, false, true);
+    }
+
+    @Override
+    protected void setChildBounds(Set<Control> controls, int weightTotal) {
+      int cw = w - (children.size() - 1) * FOLDER_MARGIN;
+      if (weightTotal <= 0) {
+        int fw = cw / children.size();
+        int rem = cw % children.size();
+        for (int i = 0, fx = x; i < children.size(); i++) {
+          int add = rem-- > 0 ? 1 : 0;
+          children.get(i).setBounds(controls, fx, y, fw + add, h);
+          children.get(i).weight = fw;
+          fx += fw + add + FOLDER_MARGIN;
+        }
+      } else {
+        int diff = 0;
+        if (weightTotal != cw) {
+          for (Element child : children) {
+            child.weight = (int)((child.weight * cw) / (double)weightTotal);
+            diff += child.weight;
+          }
+          diff = cw - diff;
+          if (diff >= children.size()) {
+            for (Element child : children) {
+              child.weight++;
+            }
+            diff -= children.size();
+          }
+        }
+        for (int i = 0, fx = x; i < children.size(); i++) {
+          int add = diff-- > 0 ? 1 : 0;
```

Comment:
Same as `rem--` : move `diff--` to its own line for the sake of clarity

Reply:
ditto